src/tactic/nat_cases.lean

```diff
@@ -0,0 +1,163 @@
+/-
+Copyright (c) 2019 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Scott Morrison
+
+Case bashing on natural numbers.
+
+`nat_cases a ≤ n < b` breaks into the following cases:
+`n < a`, one case `n = k` for each `a ≤ k < b`, and `n ≥ b`
+
+`nat_cases n`
+1) inspects hypotheses looking for "easy" upper and lower bounds on `n`,
+2) calls `nat_cases a ≤ n < b` with appropriate values of `a` and `b`,
+3) calls `linarith` to discharge the `n < a` and `n ≥ b` cases.
```

Comment:
Isn't `linarith` expensive here? If you've already found these upper and lower bounds, can't you plug in the proof directly?

Reply:
Oops, actually, we're already doing this, and the other documentation correctly describes this. I've updated the module comment.